java-squid/src/main/java/org/sonar/java/ast/parser/TreeFactory.java

```diff
@@ -1067,42 +868,35 @@ public StatementExpressionListTreeImpl newForInitDeclaration(ModifiersTreeImpl m
     }
 
     StatementExpressionListTreeImpl result = new StatementExpressionListTreeImpl(variables);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "result". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)